herbert_core/utilities/geometry/box_intersect.m

```diff
@@ -0,0 +1,156 @@
+function inter_points = box_intersect(box_minmax,cross_plain)
+% Calculate intersection points between the box in ND (2,3,4)
```

Comment:
I am reviewing this from the bottom up so these input descriptions answer some of my comments below. It would still be worth providing some description on the individual functions I mention.